cmd/export_ledgers.go

```diff
@@ -54,13 +82,62 @@ func printTransformStats(attempts, failures int, printLog bool) {
 
 	results, err := json.Marshal(resultsMap)
 	if err != nil {
-		cmdLogger.Fatal("Could not marshall results: ", err)
+		cmdLogger.Fatal("Could not marshal results: ", err)
 	}
 
-	if printLog {
-		fmt.Println(string(results))
-	} else {
-		cmdLogger.Info(string(results))
+	cmdLogger.Info(string(results))
+}
+
+func generateRunId() string {
+	return uuid.New().String()
+}
+
+func uploadToGcs(credentialsPath, bucket, runId, path string) error {
+	// Use credentials file in dev/local runs. Otherwise, derive credentials from the service account.
+	if len(credentialsPath) > 0 {
+		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", credentialsPath)
+		cmdLogger.Infof("Using credentials found at: %s", credentialsPath)
+	}
+
+	reader, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s: %v", path, err)
+	}
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Hour)
+	defer cancel()
+
+	objectPath := filepath.Join(runId, path)
+	wc := client.Bucket(bucket).Object(objectPath).NewWriter(ctx)
+
+	cmdLogger.Infof("Uploading %s to gs://%s/%s", path, bucket, objectPath)
+
+	var written int64
+	if written, err = io.Copy(wc, reader); err != nil {
+		return fmt.Errorf("unable to copy: %v", err)
+	}
+	err = wc.Close()
```

Comment:
Do we plan on deleting the local file if GCS upload is successful? Or would you rather the crawler in the `airflow-workers` remove files after `X` minutes have passed?

Reply:
I don't mind adding it, but is there an urgency to have the files deleted? Are we worried about running out of disk for unbounded runs?

Reply:
Not for the unbounded runs, but for historical loads with parallelized execution we could quickly run out of disk space if we do not spec the `airflow-workers` appropriately (set to 32GB of shared storage rn). I'm ok if we include the feature in a later PR or keep the existing `poststart.sh` script on the `airflow-workers`. That `poststart.sh` script today will clean up the old files, but in general I would like old files cleaned up to keep persisted storage small. 

Thoughts on keeping the current design, a long-running script that does the cleanup in the background of the `airflow-workers`, versus adding local file deletion to the `stellar-etl` image?

Reply:
Seems cleaner to just do it here. Added some code that cleans up files after they are uploaded to GCS. Exported files that are not uploaded are left alone.